### ISSUE TITLE
Create conversion methods that allow lifting a value or traversable type into a Monad Transformer

### ISSUE THREAD

**Issue by johnmcclean-aol** [aol/cyclops-react#145]
Either by lifting the current value / traversable inside another monad type, or by converting the elements of a monad to another monad type.

```java
a.
Maybe.of(3)
          .transformers()
          .toMaybeT(Stream::of);

b.
Maybe.of(3)
          .transformers()
          .toStreamT(Stream::of);
```

a. creates a MaybeT by wrapping the current maybe inside a Stream

b. creates a StreamT by wrapping the value 3 inside a Stream inside a Maybe.

**Comment by smillies** [comment]
Isn't that still too complicated? After all, when giving some type that I need to embed in a monad just so I can manipulate it with a transformer, will I usually care about the type of the intervening monad? Wouldn't just toXyzT() suffice, without the argument? A concern might be type-safety, but that should be addressed by issue #144.

**Comment by johnmcclean-aol** [comment]
Yes, I agree. I had implemented this, and wasn't happy with it, so have removed it. I'll give toXyzT() a try also, do you have any examples / use cases where you feel that is a good fit for?

My current thoughts are to provide something like fromValue, fromPublisher, fromIterable, fromStream, fromOptional and fromFuture static factory methods for each MonadTransformer type e.g.

```java     
      ListT<Integer> listT = ListT.fromFuture(CompletableFuture.supplyAsync(() -> asList(20, 40)));
      ListT<Integer> listT = ListT.fromValue(Maybe.just(asList(20, 40));
      ListT<Integer> listT = ListT.fromOptional(Optional.of(asList(20, 40));
      ListTValue<Integer> listT = ListT.fromIterable(javaslang/fj.Option.of(asList(20, 40));

      ListT<Integer> listT = ListTSeq.fromPublisher(Flux.just(asList(20, 40));
      ListT<Integer> listT = ListTSeq.fromStream(Stream.of(asList(20, 40));
      ListTSeq<Integer> listT = ListTSeq.fromIterable(javaslang/fj.Stream.ofAll(asList(20, 40));
```

Where ListT is a common interface that ListTValue and ListTSeq implement.

**Comment by smillies** [comment]
I was primarily looking for a way to create a ListT from a list. The factory method approach seems better than the "fluid" approach in those cases where you are given the list value. But hand-wrapping the list in an Optional as in your examples 3 and 4 may be a bit of a nuisance. Would it not be possible to have something like   `ListT<Integer> listT = ListT.fromValue(asList(20, 40));` as well?

The case may be different when not given the list, but some AnyM. It makes a difference if I write  `ListT<Integer> listT = ListT.fromAnyM(AnyM.ofSeq(asList(20, 40)));` or `ListT<Integer> listT = ListT.of(AnyM.ofSeq(asList((asList(2,4)))));` Somewhat counterintuitively (to me), it is the second form that is correct. (In the sense of working with that ["divide" example](https://medium.com/@johnmcclean/introducing-the-cyclops-monad-api-a7a6b7967f4d) from your blog, repeated below.) Do you have a proposal how to create a ListT from an AnyM? Perhaps it would be good to have _both_ the factory methods on ListT _and_ the monad specific transformers via AnyM#transformers().

As in
```
    @Test
    public void asyncFutureWithList_NewAPI() {
        ListT<Integer> dividends = ListT.fromCompletableFuture(CompletableFuture.supplyAsync(() -> asList(20, 40)));
        ListT<Integer> divisors = AnyM.ofSeq(asList(20, 40)).transformers().toListT();
        
        BiFunction<ListT<Integer>, ListT<Integer>, ListT<Integer>> divide = ListT.lift2(this::divide);
        ListT<Integer> result = divide.apply(dividends, divisors); 
        
        AnyM<List<Integer>> anyM = result.unwrap();
        CompletableFuture<List<Integer>> completed = anyM.unwrap();
        List<Integer> quotients = completed.join(); 
        assertEquals(asList(10, 5, 20, 10), quotients);
    }
```

**Comment by johnmcclean-aol** [comment]
This is useful.

I think this suggests two core use cases.

1) Where we need / want to manipulate a specific monad inside a specific monad
2) Where the type signature of a lifted function or method requires monad transformers, but we don't care (so much) about the wrapping type.

The static factory methods are probably solve 1)

In the second case, the wrapping monad type may not actually matter (or may only matter for the first parameter for example). For subsequent parameters forcing the wrapping of List inside another monad type creates unnecessary noise.

```java 
ListT<Integer> list = ListT.fromListValue(asList(20,40)); //Wrap a List in a Maybe or Optional
ListT<Integer> list = ListT.fromListSeq(asList(20,40)); //Wrap a List inside a List
```

Similarly on the types directly, we could add two toXyzT() methods

```java 
ListT<Integer> list = ListX.of(20,40).toListTValue();
ListT<Integer> list = ListX.of(20,40).toListTSeq();
```

**Comment by johnmcclean-aol** [comment]
Implemented as static methods in #190

I'll open up another issue for adding MonadTransformer methods directly to the different types to solve case 2) above.
